api/src/main/java/mcoffin/rogue/api/EncoderComponent.java

```diff
@@ -0,0 +1,6 @@
+package mcoffin.rogue.api;
+
+public interface EncoderComponent {
+  public double getEncoderPosition();
```

Comment:
Encoders just return the number of clicks. It's pretty easy to get forward or backwards distance from the reset, you just need to multiple the number of clicks by the circumference of the wheel, but I'm not sure how you want to get a 2-dimensional position from that value. You probably could, but considering how Drive isn't the only thing that could use this interface, wouldn't it make more sense to leave it as a double? It could take an Encoder as an argument, and return the position of just that Encoder. I think from there it would be possible for a Drive to calculate a position.